### DISCUSSION TITLE
Is is recommended to enable System cryptography: Use FIPS compliant algorithms for encryption, hashing, and signing policy

### DISCUSSION THREAD

**Discussion by agpt8** [HotCakeX/Harden-Windows-Security#248] [Q&A] [answered]
In GPO, there is a policy at Computer Configuration\Windows Settings\Security Settings\Local Policies\Security Options called System cryptography: Use FIPS compliant algorithms for encryption, hashing, and signing policy, which is set to disabled by default.

The details tab does make a compelling argument on why it should be enabled. The docs [however ](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-10/security/threat-protection/security-policy-settings/system-cryptography-use-fips-compliant-algorithms-for-encryption-hashing-and-signing) does not give much as it is no longer updated (but does say at the top that it applies to windows 11). It does link to another FIPS140 page but that does not say if this policy should be enabled or not.

Whats your take on this?
The page is still linked to the current [docs ](https://learn.microsoft.com/en-us/windows/security/operating-system-security/) under System security heading.

**Comment by HotCakeX** [comment]
Hi,
From what i've seen, FIPS compliant stuff are usually not up to date and don't include the latest algorithms, features etc. It could deprive users of some newer technology as a result. I don't have any relevant documents to link to at the moment but that's basically what I remember.

This [document](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-10/security/threat-protection/security-policy-settings/system-cryptography-use-fips-compliant-algorithms-for-encryption-hashing-and-signing) you linked does talk about FIPS and it's talking about all of the old insecure algorithms that the Harden Windows Security script disables. The algorithms the script enforces are all superior to things that FIPS recommends.

`TLS_RSA_WITH_3DES_EDE_CBC_SHA` or `3DES` are all super insecure

https://scanigma.com/knowledge-base
https://developers.cloudflare.com/ssl/reference/cipher-suites/recommendations/
https://github.com/ssllabs/research/wiki/SSL-and-TLS-Deployment-Best-Practices

**Comment by Harvester57** [comment, marked as answer]
In addition to what HotCakeX said, Microsoft used to have a blog post (now archived) that explained the rationale behind the mode, and when you should enable it, or not 

Tl;dr: if you do not have specific compliance reasons (such as, you ship a product that will be used by a US federal agency) to enable it, keep it disabled.

https://web.archive.org/web/20231122001052/https://techcommunity.microsoft.com/t5/microsoft-security-baselines/why-we-re-not-recommending-fips-mode-anymore/ba-p/701037

**Reply by agpt8** [reply]
Interesting! Thank you for the info!
